### PR TITLE
fix(pkg/plugin): copy plugins directly to the data directory

### LINKED ISSUE #7206: XDG cache dir can be deleted at any time
Helm 3 uses the XDG base dir specification and this is very nice. [The specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) states that `$XDG_CACHE_HOME` dir contains "user specific non-essential data files". To my surprise what is not stated is that cache dir can be cleaned up by user at any time. All caches (should) work like this.

In Helm 3.0.1, the cache dir is used for plugins and repositories and these per XDG can disappear at any time. I think plugins and repositories should be kept in `$XDG_DATA_HOME`.

### PR DESCRIPTION
**Pull request by adamreese**
Copy plugins from the cache rather than create a symlink.

fixes: #7206

### CODE REVIEW

**Review by technosophos — APPROVED** [review]
LGTM

**Comment by technosophos** [comment]
@marckhouzam You wanna take a look at this and verify whether it solves #7206?

File: pkg/plugin/installer/vcs_installer.go [hunk 1]
```diff
@@ -87,7 +88,8 @@ func (i *VCSInstaller) Install() error {
 		return ErrMissingMetadata
 	}
 
-	return i.link(i.Repo.LocalPath())
+	debug("copying %s to %s", i.Repo.LocalPath(), i.Path())
+	return fs.CopyDir(i.Repo.LocalPath(), i.Path())
```
Comment (marckhouzam) [inline pkg/plugin/installer/vcs_installer.go:92]: Need to update the description of this function, as it still mentions the symlink

**Review by marckhouzam — APPROVED** [review]
Looks good!  Two nits.

Do we need to do something about plugins that are already installed?
I believe that once this change is in, the user would still need to uninstall a plugin and re-install it for it to stop using the symlink.  So they would remain vulnerable to the cache being wiped for pre-installed plugins.

**Review by bacongobbler — APPROVED** [review]
Have you given this a quick spin with `helm plugin update`? I want to quickly make sure upgrading plugins from VCS repositories still work as expected. It relied on a `git pull`, but if I recall it performed the update from the cache path, not from the data path.

Line 56 in vcs_installer.go might be interesting

File: cmd/helm/load_plugins.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func loadPlugins(baseCmd *cobra.Command, out io.Writer) {
 		return
 	}
 
-	found, err := findPlugins(settings.PluginsDirectory)
+	found, err := plugin.FindPlugins(settings.PluginsDirectory)
```
[the remark this answers is not in the document]
Comment (adamreese) [reply cmd/helm/load_plugins.go:61]: Thanks pal!

**Comment by adamreese** [comment]
@bacongobbler tested and working

**Comment by liuming-dev** [comment]
Will this be in v3.2.1 release?

**Pull request merged**